pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -298,6 +299,38 @@ def test_iloc_setitem_bool_indexer(self, klass):
         expected = DataFrame({"flag": ["x", "y", "z"], "value": [2, 3, 4]})
         tm.assert_frame_equal(df, expected)
 
+    @pytest.mark.parametrize("dtype", ["int64", "Int64"])
+    def test_setitem_complete_columns_different_dtypes(self, dtype):
+        # GH: 20635
+        df = DataFrame({"A": ["a", "b"], "B": ["1", "2"], "C": ["3", "4"], "D": [1, 2]})
+        df.loc[:, ["B", "C"]] = df.loc[:, ["B", "C"]].astype("int64").astype(dtype)
```

Comment:
maybe define the RHS in a previous line?

Reply:
Yeah bit better to read, thx